code/socialDistribution/views.py

```diff
@@ -389,14 +390,24 @@ def author(request, author_id):
         author_type = REMOTE
 
     if author_type == LOCAL:
-            posts = author.posts.listed().get_public()  # get public posts only
+        posts = author.posts.listed().get_public()  # get public posts only
 
     else:
         # show public posts only
-        posts = InboxPost.objects.filter(
-            author=author.get_url_id(),
-            visibility=InboxPost.Visibility.PUBLIC
-        )
+
+        request_url = author.get_url_id().strip('/') + "/posts"
+
+        res_code, res_body = api_requests.get(request_url, send_basic_auth_header=True)
+        
+        if res_code == 200 and res_body:
+            for post in res_body["items"]:
+                if post:
+                    makePost(curr_user.id, post)
```

Comment:
Yo this function will add all those posts to the users inbox_posts field, which we don't wanna do. 

Maybe modify the function so that adding to inbox_posts is done outside the function, that way you can call makePost here()

ty for getting this done so quick.

Reply:
Ohh 😅 I thought it was fine since they're public posts, apologies
Will make the changes
Thank you!♥